invokeai/app/services/bulk_download/bulk_download_default.py

```diff
@@ -0,0 +1,177 @@
+        """
+        Create a zip file containing the images specified by the given image names or board id.
+        If download with the same bulk_download_id already exists, it will be overwritten.
+
+        :return: The name of the zip file.
+        """
+        zip_file_name = bulk_download_item_id + ".zip"
+        zip_file_path = self.__bulk_downloads_folder / (zip_file_name)
+
+        with ZipFile(zip_file_path, "w") as zip_file:
+            for image_dto in image_dtos:
+                image_zip_path = Path(image_dto.image_category.value) / image_dto.image_name
+                image_disk_path = self.__invoker.services.images.get_path(image_dto.image_name)
+                zip_file.write(image_disk_path, arcname=image_zip_path)
+
+        return str(zip_file_name)
+
+    # from https://stackoverflow.com/questions/7406102/create-sane-safe-filename-from-any-unsafe-string
+    def _clean_string_to_path_safe(self, s: str) -> str:
+        """Clean a string to be path safe."""
+        return "".join([c for c in s if c.isalpha() or c.isdigit() or c == " "]).rstrip()
+
+    def _signal_job_started(self, bulk_download_id: str, bulk_download_item_id: str) -> None:
+        """Signal that a bulk download job has started."""
+        if self.__event_bus:
+            assert bulk_download_id is not None
+            self.__event_bus.emit_bulk_download_started(
+                bulk_download_id=bulk_download_id,
+                bulk_download_item_id=bulk_download_item_id,
+            )
+
+    def _signal_job_completed(
+        self, bulk_download_id: str, bulk_download_item_id: str, bulk_download_item_name: str
+    ) -> None:
+        """Signal that a bulk download job has completed."""
+        if self.__event_bus:
+            assert bulk_download_id is not None
+            assert bulk_download_item_name is not None
+            self.__event_bus.emit_bulk_download_completed(
+                bulk_download_id=bulk_download_id,
+                bulk_download_item_id=bulk_download_item_id,
+                bulk_download_item_name=bulk_download_item_name,
+            )
+
+    def _signal_job_failed(self, bulk_download_id: str, bulk_download_item_id: str, exception: Exception) -> None:
+        """Signal that a bulk download job has failed."""
+        if self.__event_bus:
+            assert bulk_download_id is not None
+            assert exception is not None
+            self.__event_bus.emit_bulk_download_failed(
+                bulk_download_id=bulk_download_id,
+                bulk_download_item_id=bulk_download_item_id,
+                error=str(exception),
+            )
+
+    def stop(self, *args, **kwargs):
+        """Stop the bulk download service and delete the files in the bulk download folder."""
+        # Get all the files in the bulk downloads folder
+        files = self.__bulk_downloads_folder.glob("*")
+
+        # Delete all the files
+        for file in files:
+            file.unlink()
```

Comment:
Is there a risk of deleting all of user's output images by `glob("*")` if we initialize the service as set up in https://github.com/invoke-ai/InvokeAI/pull/5159/files#diff-f0400f2fd41cfd6706a949f0bf9f0cc3e96bf5d91cfdd87b970fe7a60fb73f27R84 ?

```     
        ### dependencies.py
        output_folder = config.output_path
        # ...
        bulk_download = BulkDownloadService(output_folder=f"{output_folder}")
```
Perhaps cleaning up of these files should be left up to the user, or they should be given an opportunity to review what's being deleted.

Reply:
Noticed that we're creating a subfolder for this: `self.__bulk_downloads_folder = self.__output_folder / "bulk_downloads"`. So the risk isn't as great. Still, if we're unlinking by glob, we should keep this dir outside of image `outputs`, to avoid any mishaps. And confirming before deleting is still recommended

Reply:
I can certainty look into temp directory. 

If I globbed *.zip, that would certainly reduce the scope of the impact. I'm not sure that anything else generates zip files.

Reply:
I changed the glob to *.zip, this should reduce the scope dramatically, no dirs should be targeted.